crates/build/src/lib.rs

```diff
@@ -215,6 +232,50 @@ fn construct_workdir(app_dir: &Path, workdir: Option<impl AsRef<Path>>) -> Resul
     Ok(cwd)
 }
 
+/// Saves the build profile to the "last build profile" file.
+/// Errors are ignored as they should not block building.
+pub fn save_last_build_profile(app_dir: &Path, profile: Option<&str>) {
+    let app_stash_dir = app_dir.join(".spin");
+    _ = std::fs::create_dir_all(&app_stash_dir);
+    let last_build_profile_file = app_stash_dir.join(LAST_BUILD_PROFILE_FILE);
```

Comment:
Small nicety: do nothing in the common case:

```suggestion
pub fn save_last_build_profile(app_dir: &Path, profile: Option<&str>) {
    let app_stash_dir = app_dir.join(".spin");
    let last_build_profile_file = app_stash_dir.join(LAST_BUILD_PROFILE_FILE);
    if profile.is_none() && !last_build_profile_file.exists() {
        return;
    }
    _ = std::fs::create_dir_all(&app_stash_dir);
```